### PR TITLE
wxGTK31: 3.1.4 -> 3.1.5

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@GrahamcOfBorg eval

**Review by jonringer — COMMENTED** [review]
might not be relevant, but probably good to tackle now:

```
error: builder for '/nix/store/kxih50jb28xn8swyf5vn9nc8bld9lrym-wxwidgets-3.0.5.drv' failed with exit code 1;
       last 10 log lines:
       > applying patch /nix/store/71qmb62m4m5zl947jx53y98jakdjzycx-0001-fix-assertion-using-hide-in-destroy.patch
       > patching file src/common/toplvcmn.cpp
       > Hunk #1 succeeded at 124 (offset 2 lines).
       > configuring
       > fixing libtool script ./src/png/ltmain.sh
       > fixing libtool script ./src/tiff/config/ltmain.sh
       > fixing libtool script ./src/jpeg/ltmain.sh
       > fixing libtool script ./src/expat/expat/conftools/ltmain.sh
       > configure flags: --disable-dependency-tracking --prefix=/nix/store/yymqxrldmabpqnr6qa9svp90905xc197-wxwidgets-3.0.5 --disable-precomp-headers --enable-mediactrl --disable-compat24 --enable-compat26 --enable-unicode --with-opengl
       > configure: error: unrecognized options: --disable-compat24, --enable-compat26
       For full logs, run 'nix log /nix/store/kxih50jb28xn8swyf5vn9nc8bld9lrym-wxwidgets-3.0.5.drv'.
```

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20544,34 +20544,17 @@ with pkgs;
...
-
-  wxGTK31-gtk3 = callPackage ../development/libraries/wxwidgets/3.1 {
-    withGtk2 = false;
-    inherit (darwin.stubs) setfile;
-    inherit (darwin.apple_sdk.frameworks) AGL Carbon Cocoa Kernel QTKit;
-  };
-
-  wxmac = callPackage ../development/libraries/wxwidgets/3.0/mac.nix { };
+  wxGTK31-gtk2 = wxGTK30.override { withGtk2 = true; };
+  wxGTK31-gtk3 = wxGTK30.override { withGtk2 = false; };
```
Comment (KoviRobi) [inline pkgs/top-level/all-packages.nix:20557]: Is this a typo that's meant to be `wxGTK31.override`?

Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:20557]: Yep, I left that escape :/

**Comment by Artturin** [comment]
this didn't have to go into staging
```
./maintainers/scripts/rebuild-amount.sh  HEAD~3
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
    156 x86_64-linux
```

**Comment by AndersonTorres** [comment]
Originally it triggered red flags about many builds on master.

**Pull request merged**